### PR TITLE
ENH: Improve ARDL and UECM

### PR DESCRIPTION
**Pull request by bashtage**
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by josef-pkt** [comment]
This PR cause a style check failure in main (that I have now also in my PR)
`statsmodels/tsa/ardl.py:0:1: E902 FileNotFoundError: [Errno 2] No such file or directory: 'statsmodels/tsa/ardl.py'`

**Comment by josef-pkt** [comment]
AFAIR, there should not be a `theta_from_tau` in copulas anymore

**Comment by bashtage** [comment]
There was a stray `_theta_from_tau` that was producing an error in the copula notebook, so I had to guess.  There are a lot of these functions in the archimedian copulas.

File: statsmodels/distributions/copula/copulas.py [hunk 1]
```diff
@@ -462,7 +462,7 @@ def fit_corr_param(self, data):
             warnings.warn("currently only first pair of data are used"
                           " to compute kendall's tau")
         tau = stats.kendalltau(x[:, 0], x[:, 1])[0]
-        return self._theta_from_tau(tau)
```
Comment (bashtage) [inline statsmodels/distributions/copula/copulas.py:465]: This was in the base class but had to implementation. There were some `theta_from_tau` methods in subclasses.

**Comment by josef-pkt** [comment]
I will go over copulas again before 0.13
In archimedean theta_from_tau makes sense as long as we have only the basic 1-parameter archimedean.
But it doesn't generalize and we need more generic names in base Copula class and non-archimedean copulas.
(subclass specific names are mainly for user convenience and easier recognition)

**Pull request merged**